app/models/manageiq/providers/hawkular/middleware_manager/alert_profile_manager.rb

```diff
@@ -0,0 +1,110 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::AlertProfileManager
+    require 'hawkular/hawkular_client'
+
+    def initialize(ems)
+      @alerts_client = ems.connect.alerts
+    end
+
+    def process_alert_profile(operation, miq_alert_profile)
+      profile_id = miq_alert_profile[:id]
+      old_alerts_ids = miq_alert_profile[:old_alerts_ids]
+      new_alerts_ids = miq_alert_profile[:new_alerts_ids]
+      old_assignments_ids = miq_alert_profile[:old_assignments_ids]
+      new_assignments_ids = miq_alert_profile[:new_assignments_ids]
+      case operation
+      when :update_alerts
+        update_alerts(profile_id, old_alerts_ids, new_alerts_ids, old_assignments_ids)
+      when :update_assignments
+        update_assignments(profile_id, old_alerts_ids, old_assignments_ids, new_assignments_ids)
+      end
+    end
+
+    def update_alerts(profile_id, old_alerts_ids, new_alerts_ids, old_assignments_ids)
+      unless old_assignments_ids.empty?
+        to_remove_alerts_ids = old_alerts_ids - new_alerts_ids
+        to_add_alerts_ids = new_alerts_ids - old_alerts_ids
+        to_remove_alerts_ids.each do |alert_id|
+          group_trigger = @alerts_client.get_single_trigger "MiQ-#{alert_id}", true
+          unassign_members(group_trigger, profile_id, old_assignments_ids)
+        end
+        to_add_alerts_ids.each do |alert_id|
+          group_trigger = @alerts_client.get_single_trigger "MiQ-#{alert_id}", true
+          assign_members(group_trigger, profile_id, old_assignments_ids)
+        end
+      end
+    end
+
+    def update_assignments(profile_id, old_alerts_ids, old_assignments_ids, new_assignments_ids)
+      to_unassign_ids = old_assignments_ids - new_assignments_ids
+      to_assign_ids = new_assignments_ids - old_assignments_ids
+      if !to_unassign_ids.empty? || !to_assign_ids.empty?
```

Comment:
Part of the problem in this case is that it's `!a.empty?` which in itself is complicated.  It might be cleaner to use `a.present?`...so that could be written as `if to_unassign_ids.present? || to_assign_ids.present?` which to me is the cleanest of all.  Alternately if you are guaranteed that those things are Arrays (and not possibly nil)...you can use `.any?`, making it `if to_unassign_ids.any? || to_assign_ids.any?`.

I find that coding in positive terminology tends to lead to the cleanest conditionals